ui/src/shared/components/TimeMachine/TimeMachine.tsx

```diff
@@ -756,12 +761,17 @@ class TimeMachine extends PureComponent<Props, State> {
     this.props.updateScript(script, this.stateToUpdate)
   }
 
-  private getASTResponse = async (script: string, update: boolean = true) => {
+  private getASTResponse = async (
+    script: string,
+    update: boolean = true
+  ): Promise<void> => {
     const {fluxLinks, updateScriptStatus, isInCEO} = this.props
+    this.setState({script})
```

Comment:
What's the difference between the `script` stored in the component state and the `script` passed in through the component's props?

Reply:
The one in props ultimately comes from redux. I stored this in state so that the script component can update prior to the AST request.